### PR TITLE
tracing: add --sentry-tags

### PR DESCRIPTION
**Pull request by guswynn**
Currently sentry tags use `--opentelemetry-resource`, which does not include region (and doesn't need to). This also means clusters miss these tags if `--opentelemetry-endpoint` isn't set. 

This adds `--sentry-tags` so I can (after this releases) add `--sentry-tag=region={region}` to environmentd's. Currently its very unclear what region an error is coming from!

### Motivation

  * This PR adds a known-desirable feature.
### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/orchestrator-tracing/src/lib.rs [hunk 1]
```diff
@@ -179,6 +179,17 @@ pub struct TracingCliArgs {
     /// See: <https://docs.sentry.io/platforms/rust/configuration/options/#environment>
     #[clap(long, env = "SENTRY_ENVIRONMENT")]
     pub sentry_environment: Option<String>,
+    /// Additional tags to send with all sentry events. These are added to
+    /// `--opentelemetry-resource`
+    ///
+    /// Requires that the `--sentry-dsn` option is specified.
+    #[clap(
+        long,
+        env = "OPENTELEMETRY_RESOURCE",
```
Comment (benesch) [inline src/orchestrator-tracing/src/lib.rs:188]:
```suggestion
        env = "SENTRY_TAG",
```

File: src/orchestrator-tracing/src/lib.rs [hunk 2]
```diff
@@ -179,6 +179,17 @@ pub struct TracingCliArgs {
     /// See: <https://docs.sentry.io/platforms/rust/configuration/options/#environment>
     #[clap(long, env = "SENTRY_ENVIRONMENT")]
     pub sentry_environment: Option<String>,
+    /// Additional tags to send with all sentry events. These are added to
+    /// `--opentelemetry-resource`
```
Comment (benesch) [inline src/orchestrator-tracing/src/lib.rs:183]: (The current phrasing makes it sound like we attach the Sentry tags to OpenTelemetry traces, which we do not!)

```suggestion
    /// Tags to send with all Sentry events, in addition to the tags
    /// specified by `--opentelemetry-resource`.
```

File: src/controller/src/clusters.rs [hunk 3]
```diff
@@ -639,6 +639,8 @@ where
                             format!("--internal-http-listen-addr={}", assigned["internal-http"]),
                             format!("--opentelemetry-resource=cluster_id={}", cluster_id),
                             format!("--opentelemetry-resource=replica_id={}", replica_id),
+                            format!("--sentry-tag=cluster_id={}", cluster_id),
+                            format!("--sentry-tag=replica_id={}", replica_id),
```
Comment (benesch) [inline src/controller/src/clusters.rs:643]: Wait, why do we have to specify this as a Sentry tag, given that it's already an OTel resource?

File: src/orchestrator-tracing/src/lib.rs [hunk 4]
```diff
@@ -179,6 +179,17 @@ pub struct TracingCliArgs {
     /// See: <https://docs.sentry.io/platforms/rust/configuration/options/#environment>
```
Comment (benesch) [inline src/orchestrator-tracing/src/lib.rs:179]: Can we adjust the docs on `--opentelemetry-resource` above to say "Also used for Sentry tags, regardless of whether `--opentelemetry-endpoint` is specified"?

File: src/orchestrator-tracing/src/lib.rs [hunk 5]
```diff
@@ -354,6 +367,8 @@ impl NamespacedOrchestrator for NamespacedTracingOrchestrator {
                             .expect("opentelemetry-header had non-ascii value"),
                     ));
                 }
+                // TODO(guswynn): If `--sentry-dsn` is passed, but not `--opentelemetry-endpoint`,
+                // these are skipped for clusters.
```
Comment (benesch) [inline src/orchestrator-tracing/src/lib.rs:371]: This seems like an easy fix—just pop this out of the `if let Some(endpoint) = opentelemetry_endpoint {` block?

File: src/controller/src/clusters.rs (hunk 3 above)
Comment (guswynn) [reply src/controller/src/clusters.rs:643]: so I can deprecate the dual-usage of `--opentelemetry-resource`

File: src/orchestrator-tracing/src/lib.rs (hunk 4 above)
Comment (guswynn) [reply src/orchestrator-tracing/src/lib.rs:179]: that seems fine as well

File: src/controller/src/clusters.rs (hunk 3 above)
Comment (guswynn) [reply src/controller/src/clusters.rs:643]: see below, we can avoid this

**Comment by guswynn** [comment]
@benesch updated with your suggestions!

**Review by benesch — APPROVED** [review]

**Comment by benesch** [comment]
Thank you!

**Pull request merged**